packages/components/src/circular-option-picker/index.tsx

```diff
@@ -7,35 +7,50 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { Icon, check } from '@wordpress/icons';
+import { createContext, useContext } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Button from '../button';
+import {
+	Composite,
+	CompositeGroup,
+	CompositeItem,
+	useCompositeState,
+} from '../composite';
 import Dropdown from '../dropdown';
 import Tooltip from '../tooltip';
 import type {
 	CircularOptionPickerProps,
 	DropdownLinkActionProps,
+	OptionGroupProps,
 	OptionProps,
 } from './types';
 import type { WordPressComponentProps } from '../ui/context';
 import type { ButtonAsButtonProps } from '../button/types';
 
+const CircularOptionPickerContext = createContext( {} );
+
 export function Option( {
 	className,
 	isSelected,
 	selectedIconProps,
 	tooltipText,
 	...additionalProps
 }: OptionProps ) {
-	const optionButton = (
-		<Button
+	const compositeState = useContext( CircularOptionPickerContext );
+
+	const optionControl = (
+		<CompositeItem
+			as={ Button }
+			className={ 'components-circular-option-picker__option' }
 			isPressed={ isSelected }
```

Comment:
I think we need to change it from `aria-pressed` to `aria-selected`?

We may also want to make sure a consumer can add `aria-multiselectable` to the `listbox`.